lib/logstash/codecs/multiline.rb

```diff
@@ -180,45 +237,62 @@ def accept(listener)
     end
   end
 
-  def decode(text, &block)
-    text = @converter.convert(text)
-    text.split("\n").each do |line|
-      match = @grok.match(line)
-      @logger.debug("Multiline", :pattern => @pattern, :text => line,
-                    :match => !match.nil?, :negate => @negate)
+  def buffer(line)
+    @buffer_bytes += line.bytesize
+    @buffer.push(line)
+  end
 
-      # Add negate option
-      match = (match and !@negate) || (!match and @negate)
-      @handler.call(line, match, &block)
-    end
-  end # def decode
+  def reset_buffer
+    @buffer = []
+    @buffer_bytes = 0
+  end
+
+  # TODO: (colin) this method is not clearly documented as being required by the AutoFlush class & tasks.
+  # I belive there is a problem here with the usage of @auto_flush_block which assumes to be in the
```

Comment:
This auto_flush method is available to be called by another object (input, pipeline or component) if it wished to.
At the moment the AutoFlush class calls it in a retriggerable timer.